src/modules/esl-utils/async/promise/event.ts

```diff
@@ -0,0 +1,48 @@
+/** Interface to describe abstract listenable target */
+export type ListenableTarget = {
+  addEventListener: (
+    event: string,
+    callback: (payload: any) => void,
+    options?: boolean | AddEventListenerOptions | undefined
+  ) => void;
+  removeEventListener: (
+    event: string,
+    callback: (payload: any) => void,
+    options?: boolean | AddEventListenerOptions | undefined
+  ) => void;
+};
+
+/**
+ * @returns Promise that will be resolved by dispatching `event` on `target`
+ * Or it will be rejected in `timeout` if it's specified
+ * Optional `options` for addEventListener can be also specified
+ */
+export function promisifyEvent(
+  target: ListenableTarget,
+  event: string,
+  timeout?: number | null | undefined,
+  options?: boolean | AddEventListenerOptions
+): Promise<Event> {
+  return new Promise((resolve, reject) => {
+    function eventCallback(e: Event): void {
+      target.removeEventListener(event, eventCallback, options);
+      resolve(e);
+    }
+
+    target.addEventListener(event, eventCallback, options);
+    if (typeof timeout === 'number' && timeout >= 0) {
+      setTimeout(() => reject(new Error('Rejected by timeout')), timeout);
```

Comment:
It lacks an unsubscribe from the event

Reply:
Issue created: https://github.com/exadel-inc/esl/issues/1698

Reply:
Issue created: https://github.com/exadel-inc/esl/issues/1698